test/file_xfr_test.go

```diff
@@ -0,0 +1,101 @@
+package test
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/coredns/coredns/plugin/test"
+
+	"github.com/miekg/dns"
+)
+
+func TestLargeAXFR(t *testing.T) {
+	t.Parallel()
+
+	// Build a large zone in text format.  It contains 64K AAAA RRs.
+	var sb strings.Builder
+	const numAAAAs = 65536
+	sb.WriteString("example.com. IN SOA . . 1 60 60 60 60\n")
+	sb.WriteString("example.com. IN NS ns.example.\n")
+	for i := 0; i < numAAAAs; i++ {
+		sb.WriteString(fmt.Sprintf("%d.example.com. IN AAAA 2001:db8::1\n", i))
+	}
+
+	// Setup the zone file and CoreDNS to serve the zone, allowing zone transfer
+	name, rm, err := test.TempFile(".", sb.String())
+	if err != nil {
+		t.Fatalf("Failed to create zone: %s", err)
+	}
+	defer rm()
+
+	corefile := `example.com:0 {
+       file ` + name + ` {
+           transfer to *
+       }
+}
+`
+
+	// Start server, and send an AXFR query to the TCP port.  We set the deadline to prevent the test from hanging.
+	i, _, tcp, err := CoreDNSServerAndPorts(corefile)
+	if err != nil {
+		t.Fatalf("Could not get CoreDNS serving instance: %s", err)
+	}
+	defer i.Stop()
```

Comment:
```suggestion
```

Reply:
hmm...this may not be a big deal anyway as it's just a test, but is there anything wrong about guaranteeing to call `Stop` using `defer`?